internal/resources/syntheticmonitoring/resource_probe.go

```diff
@@ -163,14 +163,31 @@ func ResourceProbeUpdate(ctx context.Context, d *schema.ResourceData, meta inter
 
 func ResourceProbeDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	c := meta.(*common.Client).SMAPI
-	var diags diag.Diagnostics
 	id, _ := strconv.ParseInt(d.Id(), 10, 64)
-	err := c.DeleteProbe(ctx, id)
+
+	// Remove the probe from any checks that use it.
+	checks, err := c.ListChecks(ctx)
 	if err != nil {
 		return diag.FromErr(err)
 	}
+	for _, check := range checks {
+		for i, probeID := range check.Probes {
+			if probeID == id {
+				check.Probes = append(check.Probes[:i], check.Probes[i+1:]...)
+				if len(check.Probes) == 0 {
```

Comment:
nit: We could potentially move this verification up before doing the re-slice.

Reply:
Done!